kit/inputs/src/UITextView/UITextView.tsx

```diff
@@ -0,0 +1,109 @@
+import { useAutogrow, useAutoFocus, useHandleRef } from './hooks';
+import type { UITextViewProps, UITextViewRef } from './types';
+
+/**
+ * The input on the ios platform has a feature -
+ * its height in `multiline={true}` mode is greater than in `multiline={false}`.
+ * Experimentally, it was found that it increases from above by 4pt, and from below by 1pt.
+ */
+export const EXTRA_TOP_PADDING_OF_MULTILINE_INPUT = Platform.OS === 'ios' ? 4 : 0;
+export const EXTRA_BOTTOM_PADDING_OF_MULTILINE_INPUT = Platform.OS === 'ios' ? 1 : 0;
+
+const textViewTypographyVariant = TypographyVariants.ParagraphText;
+const textViewLineHeight =
+    StyleSheet.flatten(Typography[textViewTypographyVariant]).lineHeight ??
+    UILayoutConstant.smallCellHeight;
+
+export const UITextView = React.forwardRef<UITextViewRef, UITextViewProps>(
+    function UITextViewForwarded(
+        {
+            placeholderTextColor = ColorVariants.TextSecondary,
+            style,
+            noPersonalizedLearning,
+            onHeightChange,
+            maxNumberOfLines,
+            onNumberOfLinesChange,
+            ...textInputProps
+        }: UITextViewProps,
+        passedRef,
+    ) {
+        const { autoFocus, onChange: onChangeProp, multiline } = textInputProps;
+
+        const textInputRef = React.useRef<TextInput>(null);
+        const theme = useTheme();
+        const autoFocusProp = useAutoFocus(textInputRef, autoFocus);
+
+        const { onChange, remeasureInputHeight, numberOfLines, autogrowStyle } = useAutogrow(
+            textInputRef,
+            textViewLineHeight,
+            onChangeProp,
+            multiline,
+            maxNumberOfLines,
+            onHeightChange,
+            onNumberOfLinesChange,
+        );
+
+        useHandleRef(textInputRef, passedRef, remeasureInputHeight);
+
+        const onLayout = React.useCallback(
+            function onLayout(event) {
+                textInputProps.onLayout?.(event);
+                remeasureInputHeight?.();
+            },
+            [remeasureInputHeight, textInputProps],
+        );
+
+        return (
+            <TextInput
+                ref={textInputRef}
+                {...textInputProps}
+                autoFocus={autoFocusProp}
+                // This is our custom prop, we do it in native for Android
+                {...(Platform.OS === 'android' ? { noPersonalizedLearning } : null)}
+                placeholderTextColor={theme[placeholderTextColor]}
+                selectionColor={theme[ColorVariants.TextAccent]}
+                // @ts-ignore
+                keyboardAppearance={theme[ColorVariants.KeyboardStyle]}
+                underlineColorAndroid="transparent"
+                style={[
+                    styles.input,
+                    style,
+                    {
+                        color: theme[ColorVariants.TextPrimary],
+                        paddingTop: multiline ? -EXTRA_TOP_PADDING_OF_MULTILINE_INPUT : 0,
```

Comment:
What do you think to extract it to `styles`, sth like:
```ts
const styles = StyleSheet.create({
  inputMultiline: Platform.select({ ios: { paddingTop: -4, paddingBottom: -1 }, default: {}),
})
```
and then use it like `multiline ? styles. inputMultiline : null`
?